src/components/Input/searchInput.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { SearchIcon } from '../../assets';
+import IconWrap from '../../components/ui/svgWrapper';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks';
+import { setSearch } from '../../redux/slice/querySearch';
+import useDebounce from '../hooks';
+const SearchInput = () => {
+  const [value, setValue] = useState('');
+  const { search } = useAppSelector((state) => state.query);
```

Comment:
Why exactly are you saving search value to redux store?

Reply:
The were not on the same file so I didn't want to pass it down as props across their respective children